### ISSUE TITLE
Update architecture docs based on "Life of a Query" tech note

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#10795]
Richard Loveland (rmloveland) commented:

We have recently (June 2021) seen some [very nice updates](https://github.com/cockroachdb/cockroach/commit/706eadfe7833662afbf116d9425909e83a1eef21) to the ["life of a query" tech note in the CRDB repo](https://github.com/cockroachdb/cockroach/blob/master/docs/tech-notes/life_of_a_query.md).

We should review the content of "life of a query" and port those parts of it that are appropriate for end users to the relevant sections of the [CockroachDB architecture docs](https://www.cockroachlabs.com/docs/stable/architecture/overview.html).

See also:

- cockroachdb/docs#5467
- cockroachdb/docs#10794

Jira Issue: DOC-1541

**Comment by ianjevans** [comment]
As part of these updates, it'd be great to have a more focused discussion of our requirements and recommendations for the number of replicas, the number of availability zones, and the number of nodes within a cluster.

From Slack:
> A customer asked me today if we have any docs that point out that it's a best practice to use an odd number of replicas (3, 5, 7, etc.) -- and I can't find one
I think there should be one.

This has also come up as a hole in our architecture in CockroachCloud, when creating a multi-region cluster. We have a couple places where we assert the minimum or recommended values within regions and zones but don't really have a higher-level doc that explains why those minimums/recommendations are present.

**Comment by rmloveland** [comment]
Makes sense, thanks for capturing this Ian!  Also relates to https://github.com/cockroachdb/docs/issues/9239

**Comment by mwang1026** [comment]
Related:
https://github.com/cockroachdb/docs/issues/5467
https://github.com/cockroachdb/docs/issues/4279
